File: 2401-count-asterisks/count-asterisks.py
class Solution(object):
    def countAsterisks(self, s):
        """
        :type s: str
        :rtype: int
        """
        asterisk_count = 0
        inside_pair = False

        for char in s:
            if char == '|':
                inside_pair = not inside_pair
            elif char == '*' and not inside_pair:
                asterisk_count += 1
        
        return asterisk_count

File: 2401-count-asterisks/count-asterisks.js
/**
 * @param {string} s
 * @return {number}
 */
var countAsterisks = function(s) {
    let asterisk_count  = 0;
    let inside_pair = false;

    for(char of s){
        if(char === "|"){
            inside_pair = !inside_pair;
        }
        if(char === "*" && !inside_pair){
            asterisk_count += 1;
        }
    }

    return asterisk_count;
};